lightgbm/src/test/scala/com/microsoft/azure/synapse/ml/lightgbm/split1/LightGBMTestUtils.scala

```diff
@@ -114,6 +114,6 @@ trait LightGBMTestUtils extends TestBase {
   val validationCol = "validation"
   val seed = 42L
 
-  lazy val executionMode: String = LightGBMConstants.StreamingExecutionMode
-  lazy val executionModeSuffix = ", " + executionMode
+  val dataTransferMode: String = LightGBMConstants.StreamingDataTransferMode
```

Comment:
Do you really need this val? Why not use `LightGBMConstants.StreamingDataTransferMode` directly?

Reply:
Yes, this is needed. It is overridden in some test classes to be a different value. We have suites that test everything for "streaming" and then another suite that just runs the same tests overriding with "bulk". Is there a way to make this more obvious? (in C# this would be done with "vritual" keyword to indicate it can be overridden)